Add paragraph re: funding (w/ Stefan) (#246)
scipy/scipy-articles: scipy-1.0/paper.tex

high-quality contributions.

\subsection*{Funding}

%100,000 annual salary, 112 person-years, 414,473 lines of code -> 11,192,622 (https://www.openhub.net/p/scipy/estimated\_cost, which uses the Basic COCOMO model).

The development cost of SciPy is estimated in excess of 10M dollars by
Open Hub\cite{SciPyValue}.  Yet, the project is largely unfunded, having been
developed predominantly by graduate students, faculty, and members of industry
in their free time. Small amounts of funding have been applied with success:
some meetings were sponsored by universities and industry, Google's Summer of Code
program supported infrastructure and algorithmic work, and teaching grant funds were
used early on to develop documentation.  However, these small amounts of
sporadic funding are not commensurate with the enormous stack of important scientific
software, like those that are used to plan national space missions, that rely
on SciPy.
With traditional grant mechanisms, it is easier to buy software licenses than to support the diverse development effort needed to develop open source software—a challenge that we hope to see national funding agencies address.

> The scientific Python stack is also playing an increasingly important role in industry, and while there are some encouraging signals such as TideFlow, that sector still spends comparatively little on community-developed software.

Reviewer: I think "TideFlow" should be "Tidelift"?

Reviewer 2: Woops, yes.

Reviewer 3: I would put it, "it is easier to obtain funding to buy software licenses than to gain funding to support developing open source software".

Author: I removed this text.  @charris does the revised paragraph look OK?

> For open source projects to remain a healthy underpinning infrastructure for international scientific and industrial endeavors, it needs more diverse spending to support planning, development, management, and infrastructure.

Reviewer 4: to avoid repeating "infrastructure", perhaps use "underpinning" as a noun rather than adjectival noun.
```suggestion
For open source projects to remain a healthy underpinning for international scientific and industrial endeavors, it needs more diverse spending to support planning, development, management, and infrastructure.
```

are selected on a rolling basis from contributors with a significant history of
high-quality contributions.

\subsection*{Funding}

> %100,000 annual salary, 112 person-years, 414,473 lines of code -> 11,192,622 (https://www.openhub.net/p/scipy/estimated\_cost, which uses the Basic COCOMO model).

Reviewer: cool!

Reviewer 4: Yeah, this is a great figure to have.
```
@online{SciPyValue,
author = {{Black Duck Software, Inc.}},
title = {The {SciPy} Open Source Project on {O}pen {H}ub : Estimated Cost Page},
year = 2019,
url = {https://www.openhub.net/p/scipy/estimated_cost},
urldate = {2019-09-11}
}
```

The development cost of SciPy is estimated in excess of 10M dollars by

> OpenHub.  Yet, the project is largely unfunded, having been

Reviewer 4: ```suggestion
Open Hub.  Yet, the project is largely unfunded, having been
```

developed predominantly by graduate students, faculty, and members of industry
in their free time. Small amounts of funding have been applied with success:
some meetings were sponsored by universities and industry, Google's Summer of Code
program supported infrastructure and algorithmic work, and teaching grant funds were
used early on to develop documentation.  However, these small amounts of
sporadic funding are not commensurate with the enormous stack of important scientific

> software, like those that are used to plan national space missions, that rely

Reviewer 4: I think this refers to STScI mentioned earlier, but should probably repeat the citation.

on SciPy.  In the traditional academic funding model, it is easy to support
commercial software, but hard to provide funds for community driven projects,

> and as more science depends on open source software national funding agencies

Reviewer 4: ```suggestion
and as more science depends on open source software, national funding agencies
```

urgently need to pay attention to this challenge. And, unlike for commercial
software where a license seat encompasses all development (and other) costs, agencies will

> have to tolerate the more diverse spending required to do planning and maintain

Reviewer 4: Does this mean that it is a very clear line item when funding seats of commercial software, and it is not so easy to write on paper what the costs are for open source projects?

Author: Not so much that it is hard to write down the costs as it is hard to get funding for them.  For instance, NIH commonly provides funding to support matlab seat licenses for neuroimaging research, but has only provided minimal funds for numpy, scipy, matplotlib, etc.  I don't have hard numbers, but I suspect the US tax payers have spent significantly more on matlab than on the scientific Python stack.  There are various reasons for that, but I think we only need to mention the fact here.
